charts/consul/templates/telemetry-collector-deployment.yaml

```diff
@@ -0,0 +1,362 @@
+        "consul.hashicorp.com/connect-k8s-version": {{ $.Chart.Version }}
+        {{- if .Values.telemetryCollector.customExporterConfig }}
+        # configmap checksum
+        "consul.hashicorp.com/config-checksum": {{ include (print $.Template.BasePath "/telemetry-collector-configmap.yaml") . | sha256sum }}
+        {{- end }}
+        # vault annotations
+        {{- if (and .Values.global.secretsBackend.vault.enabled .Values.global.tls.enabled) }}
+        "vault.hashicorp.com/agent-init-first": "true"
+        "vault.hashicorp.com/agent-inject": "true"
+        "vault.hashicorp.com/role": {{ .Values.global.secretsBackend.vault.consulCARole }}
+        "vault.hashicorp.com/agent-inject-secret-serverca.crt": {{ .Values.global.tls.caCert.secretName }}
+        "vault.hashicorp.com/agent-inject-template-serverca.crt": {{ template "consul.serverTLSCATemplate" . }}
+        {{- if and .Values.global.secretsBackend.vault.ca.secretName .Values.global.secretsBackend.vault.ca.secretKey }}
+        "vault.hashicorp.com/agent-extra-secret": "{{ .Values.global.secretsBackend.vault.ca.secretName }}"
+        "vault.hashicorp.com/ca-cert": "/vault/custom/{{ .Values.global.secretsBackend.vault.ca.secretKey }}"
+        {{- end }}
+        {{- if .Values.global.secretsBackend.vault.agentAnnotations }}
+        {{ tpl .Values.global.secretsBackend.vault.agentAnnotations . | nindent 8 | trim }}
+        {{- end }}
+        {{- end }}
+
+      labels:
+        consul.hashicorp.com/connect-inject-managed-by: consul-k8s-endpoints-controller
+        app: {{ template "consul.name" . }}
+        chart: {{ template "consul.chart" . }}
+        release: {{ .Release.Name }}
+        component: consul-telemetry-collector
+        {{- if .Values.global.extraLabels }}
+          {{- toYaml .Values.global.extraLabels | nindent 8 }}
+        {{- end }}
+    spec:
+      serviceAccountName: {{ template "consul.fullname" . }}-telemetry-collector
+      initContainers:
+      - name: consul-connect-inject-init
+        env:
+          - name: POD_NAMESPACE
+            valueFrom:
+              fieldRef:
+                fieldPath: metadata.namespace
+          - name: POD_NAME
+            valueFrom:
+              fieldRef:
+                fieldPath: metadata.name
+          - name: NODE_NAME
+            valueFrom:
+              fieldRef:
+                fieldPath: spec.nodeName
+          {{- if .Values.global.acls.manageSystemACLs }}
+            # The auth method should be k8s-component-auth-method but we would need to modify the acl-init job. We'll follow up with that
+          - name: CONSUL_LOGIN_AUTH_METHOD
+            {{- if and .Values.global.federation.enabled .Values.global.federation.primaryDatacenter }}
+            value: {{ template "consul.fullname" . }}-k8s-component-auth-method-{{ .Values.global.datacenter }}
+            {{- else }}
+            value: {{ template "consul.fullname" . }}-k8s-component-auth-method
+            {{- end }}
```

Comment:
Yeah if it needs perms beyond what service identity provides, it would need to use the component one. You'd need to add a role and binding rule for this to work though. So switching the method alone won't just work.

Reply:
There are changes to the `acl-init` job included in the PR to create the role and rolebinding with the anonymous rules (but not the anonymous policy ID).